waiter/src/waiter/reporter.clj

```diff
@@ -87,15 +100,53 @@
                s/Any s/Any}
               (merge {:pickled? true} config)))
 
+(defn- make-reporting-name
+  "Concatenates elements to form a dotted name for reporting, eliding any null values or empty strings."
+  [prefix & args]
+  (MetricRegistry/name prefix (into-array String args)))
+
+(let [df (DecimalFormat. "#.####################")]
+  (defn- format-value
+    "Convert numbers to string to send to graphite server"
+    [value]
+    (if (number? value)
+      (.format df value)
+      (str value))))
+
+(defn- report-to-graphite-helper
+  "Recursively traverse metrics map and send to graphite"
+  [prefix map ^GraphiteSender graphite timestamp]
+  (if (map? map)
+    (doseq [[k v] map]
+      (report-to-graphite-helper (make-reporting-name prefix (str k)) v graphite timestamp))
+    (.send graphite prefix (format-value map) timestamp)))
+
+(defn- report-to-graphite
+  "Report values from a codahale MetricRegistry"
+  [^MetricRegistry registry prefix ^MetricFilter filter ^GraphiteSender graphite]
+  (let [timestamp (.getTime (Clock/defaultClock))
+        map (metrics/metric-registry->map registry filter)]
+    (try
+      (when-not (.isConnected graphite)
+        (.connect graphite))
+      (report-to-graphite-helper prefix map graphite timestamp)
+      (.flush graphite)
+      (catch IOException e
+        (try
+          (.close graphite)
+          (catch IOException e
```

Comment:
Should this be catch throwable so that if this throws some other exception, we do not miss reporting the initial exception `e`?